script/tool/test/pubspec_check_command_test.dart

```diff
@@ -633,6 +633,126 @@ ${_topicsSection()}
       );
     });
 
+    test('fails when topic a topic name contains a space', () async {
+      final RepositoryPackage plugin =
+          createFakePlugin('plugin', packagesDir, examples: <String>[]);
+
+      plugin.pubspecFile.writeAsStringSync('''
+${_headerSection('plugin')}
+${_environmentSection()}
+${_flutterSection(isPlugin: true)}
+${_dependenciesSection()}
+${_devDependenciesSection()}
+${_topicsSection(<String>['plugin a'])}
+''');
+
+      Error? commandError;
+      final List<String> output = await runCapturingPrint(
+          runner, <String>['pubspec-check'], errorHandler: (Error e) {
+        commandError = e;
+      });
+
+      expect(commandError, isA<ToolExit>());
+      expect(
+        output,
+        containsAllInOrder(<Matcher>[
+          contains('Invalid topic value "plugin a" in "topics" section. '
+              'Topics must consist of lowercase alphanumerical characters or dash (but no double dash), '
+              'starting with a-z and ending with a-z or 0-9.'),
```

Comment:
You could just have one test for "invalid topic has the expected failure message" with the full text we want to check for, and then the other tests can only check for `contains` the first sentence of the error.